tests/h/views/api_groups_test.py

```diff
@@ -8,49 +8,86 @@
 from pyramid.httpexceptions import HTTPNoContent, HTTPBadRequest
 
 from h.views import api_groups as views
-from h.services.profile_group import ProfileGroupService
+from h.services.list_groups import ListGroupsService
 from h.services.group import GroupService
 
 
-@pytest.mark.usefixtures('profile_group_service')
 class TestGroups(object):
-    def test_groups_proxies_to_service(self, pyramid_request, profile_group_service):
-        views.groups(pyramid_request)
 
-        profile_group_service.all.assert_called_once()
+    def test_all_groups_proxies_to_service(self, anonymous_request, list_groups_service):
+        views.groups(anonymous_request)
 
-    def test_groups_passes_authority_parameter(self, pyramid_request, profile_group_service):
-        pyramid_request.params = {'authority': 'foo.com'}
+        list_groups_service.all_groups.assert_called_once_with(
+            user=None,
+            authority=None,
+            document_uri=None
+        )
 
-        views.groups(pyramid_request)
+    def test_all_groups_proxies_authority_param(self, anonymous_request, list_groups_service):
+        anonymous_request.params['authority'] = 'foo.com'
+        views.groups(anonymous_request)
 
-        c_args, c_kwargs = profile_group_service.all.call_args
-        assert c_kwargs['authority'] == 'foo.com'
+        list_groups_service.all_groups.assert_called_once_with(
+            user=None,
+            authority='foo.com',
+            document_uri=None
+        )
 
-    def test_groups_passes_document_uri_parameter(self, pyramid_request, profile_group_service):
-        pyramid_request.params = {'document_uri': 'foo.example.com'}
+    def test_all_groups_proxies_document_uri_param(self, anonymous_request, list_groups_service):
+        anonymous_request.params['document_uri'] = 'http://example.com/thisthing.html'
+        views.groups(anonymous_request)
 
-        views.groups(pyramid_request)
+        list_groups_service.all_groups.assert_called_once_with(
+            user=None,
+            authority=None,
+            document_uri='http://example.com/thisthing.html'
+        )
 
-        c_args, c_kwargs = profile_group_service.all.call_args
-        assert c_kwargs['document_uri'] == 'foo.example.com'
+    def test_uses_presenter_for_formatting(self, anonymous_request, open_groups, list_groups_service, presenter):
+        list_groups_service.all_groups.return_value = open_groups
+
+        views.groups(anonymous_request)
+
+        assert presenter.call_count == 2
+
+    def test_returns_formatted_groups(self, anonymous_request, open_groups, list_groups_service, presenter):
+        list_groups_service.all_groups.return_value = open_groups
+        presenter.asdict.return_value = {'foo': 'bar'}
+
+        result = views.groups(anonymous_request)
+
+        assert result == [presenter(group, None).asdict() for group in open_groups]
```

Comment:
This just occurred to me: if `GroupJSONPresenter` could accept a list of groups instead of just one and return a list of dicts from a single `asdicts()` method, it might be both easier to use for what the view wants to do and easier to test as you don't have multiple `asdict()` calls and return values to assert about

Reply:
@seanh All of the other presenters we have operate on single objects, which is "how I got here." But I agree, it's wonky. I do think we need a `GroupJSONPresenter` that can operate on single objects, still, but I'm wondering if it would be OK to do something like in this gist here: https://gist.github.com/lyzadanger/81138dffeb14804aede238609159e33a ? Would something like that be a plausible approach?

Reply:
Sure! That should work